src/dynamics/time_integration.jl

```diff
@@ -220,6 +229,29 @@ function leapfrog!(
     return nothing
 end
 
+function first_timesteps!(simulation::AbstractSimulation)
+    progn, diagn, model = unpack(simulation)
+    (; time_stepping) = model
+    (; Δt) = time_stepping
+
+    # decide whether to start with 1x Euler then 1x Leapfrog at Δt
+    if time_stepping.start_with_euler
+        first_timesteps!(progn, diagn, model)
+        time_stepping.start_with_euler = false      # disable for subsequent restarts
+        
+    else    # or continue with leaprog steps at 2Δt (e.g. restart)
+            # but make sure that implicit solver is initialized in that situation
+        initialize!(model.implicit, 2Δt, diagn, model)
```

Comment:
This does nothing if it's already initialised, correct?

Reply:
Yes we have a line like

```julia
    # option to skip reinitialization at restart
    (implicit.initialized && !implicit.reinitialize) && return nothing
```

now for `ImplicitPrimitiveEquation`.

Reply:
So you can force it to reinitialize but normally skipped. Only the ImplicitShallowWater is always "reinitialized" as it just needs to precompute one constant